readthedocs/projects/tasks/utils.py

```diff
@@ -154,6 +161,115 @@ def send_external_build_status(version_type, build_pk, commit, status):
+    name = "deprecated_config_file_used"
+    context_object_name = "project"
+    subject = "Add a configuration file to your project to continue building"
+    level = REQUIREMENT
+
+    def send(self):
+        """Method overwritten to remove on-site backend."""
+        backend = EmailBackend(self.request)
+        backend.send(self)
+
+
+@app.task(queue="web")
+def deprecated_config_file_used_notification():
+    """
+    Create a notification about not using a config file for all the maintainers of the project.
+
+    This is a scheduled task to be executed on the webs.
+    Note the code uses `.iterator` and `.only` to avoid killing the db with this query.
+    Besdies, it excludes projects with enough spam score to be skipped.
+    """
+    # Skip projects with a spam score bigger than this value.
+    # Currently, this gives us ~250k in total (from ~550k we have in our database)
+    spam_score = 300
+
+    projects = set()
+    start_datetime = datetime.datetime.now()
+    queryset = Project.objects.exclude(users__profile__banned=True)
+    if not settings.ALLOW_PRIVATE_REPOS:
+        # Take into account spam score on community
+        queryset = queryset.annotate(spam_score=Sum("spam_rules__value")).filter(
+            Q(spam_score__lt=spam_score) | Q(is_spam=False)
+        )
+    queryset = queryset.only("slug", "default_version")
+
+    for project in queryset.iterator():
+        # NOTE: instead of iterating over all the active versions,
+        # we can only consider the default one
+        for version in (
+            project.versions.filter(Q(active=True) | Q(slug=project.default_version))
+            .order_by("-modified")
+            # Add a limit of 15 to protect ourselves against projects with
+            # hundred of active versions
+            .only("id")[:15]
+            .iterator()
+        ):
+            build = version.builds.filter(success=True).only("_config").first()
+            if build and build.deprecated_config_used():
+                projects.add(project.slug)
+                # Do not continue iterating over the
+                # other versions when we know this project
+                # will get the notification
+                break
+
+    n_projects = len(projects)
+    log.info(
+        "Sending deprecated config file notification.",
+        query_seconds=(datetime.datetime.now() - start_datetime).seconds,
+        projects=n_projects,
+    )
+
+    projects = Project.objects.filter(slug__in=projects).order_by("id")
+    start_datetime = datetime.datetime.now()
+    for i, project in enumerate(projects.iterator()):
+
+        if i % 500 == 0:
+            log.info(
+                "Sending deprecated config file notifications.",
+                progress=f"{i}/{n_projects}",
+                current_project_pk=project.pk,
+                current_project_slug=project.slug,
+            )
+
+        users = AdminPermission.owners(project)
+        for user in users:
```

Comment:
⚠️ Review this to not send multiple emails to the same users about different projects. Let's try to send only one email per user, instead of one email per project per user.